### PR TITLE
Space Selection Fixes & Improvement while creating pin/event

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
fixes, https://github.com/acterglobal/a3/issues/2209

- Fixes issue related to space selection
- If space section is missing then show space selector rather than showing an error

Reference Video:

https://github.com/user-attachments/assets/2f1acecb-b7f0-4166-835b-1f917be8b34e

### CODE REVIEW

File: app/lib/router/general_router.dart [hunk 1]
```diff
@@ -122,10 +122,12 @@ final generalRoutes = [
     name: Routes.createPin.name,
     path: Routes.createPin.route,
     pageBuilder: (context, state) {
+      final String? spaceId = state.uri.queryParameters['spaceId'];
       return NoTransitionPage(
         key: state.pageKey,
         child: CreatePinPage(
-          initialSelectedSpace: state.uri.queryParameters['spaceId'],
+          initialSelectedSpace:
+              spaceId != null && spaceId.isNotEmpty ? spaceId : null,
```
Comment (kumarpalsinh25) [inline app/lib/router/general_router.dart:130]: Empty `spaceId` was causing issue in space selector.

File: app/lib/features/events/pages/create_event_page.dart [hunk 2]
```diff
@@ -408,9 +409,10 @@ class CreateEventPageConsumerState extends ConsumerState<CreateEventPage> {
 
   // Create event handler
   Future<void> _handleCreateEvent() async {
-    if (!_formKey.currentState!.validate()) return;
+    String? spaceId = ref.read(selectedSpaceIdProvider);
+    spaceId ??= await selectSpace();
```
Comment (gnunicorn) [inline app/lib/features/events/pages/create_event_page.dart:413]: wait ... what sorcery is this? does this ... it means .. `??=` if null run assign to? Wow .... :exploding_head:

File: app/lib/features/pins/pages/create_pin_page.dart [hunk 3]
```diff
@@ -376,6 +381,17 @@ class _CreatePinConsumerState extends ConsumerState<CreatePinPage> {
     }
   }
 
+  Future<String?> selectSpace() async {
```
Comment (gnunicorn) [inline app/lib/features/pins/pages/create_pin_page.dart:384]: so considering that this comes in here twice, this should probably be moved into a helper action, should it not?

File: app/lib/features/events/pages/create_event_page.dart [hunk 4]
```diff
@@ -477,4 +481,15 @@ class CreateEventPageConsumerState extends ConsumerState<CreateEventPage> {
       );
     }
   }
+
+  Future<String?> selectSpace() async {
+    final newSelectedSpaceId = await selectSpaceDrawer(
+      context: context,
+      currentSpaceId: ref.read(selectedSpaceIdProvider),
+      canCheck: 'CanPostPin',
```
Comment (gnunicorn) [inline app/lib/features/events/pages/create_event_page.dart:489]: and that helper should have an explicit `canCheck`-param ... as this copy-paste lead to using the wrong key ;) 

```suggestion
      canCheck: 'CanPostEvent',
```

File: app/lib/router/general_router.dart (hunk 1 above)
Comment (gnunicorn) [reply app/lib/router/general_router.dart:130]: I do wonder if it is more readible to do:
```suggestion
              (spaceId?.isNotEmpty == true) ? spaceId : null,
```

or not .. remain unsure...

File: app/lib/router/shell_routers/home_shell_router.dart [hunk 5]
```diff
@@ -483,10 +483,12 @@ final homeShellRoutes = [
       if (extra != null && extra is CalendarEvent) {
         templateEvent = extra;
       }
+      final String? spaceId = state.uri.queryParameters['spaceId'];
       return NoTransitionPage(
         key: state.pageKey,
         child: CreateEventPage(
-          initialSelectedSpace: state.uri.queryParameters['spaceId'],
+          initialSelectedSpace:
+              spaceId != null && spaceId.isNotEmpty ? spaceId : null,
```
Comment (gnunicorn) [inline app/lib/router/shell_routers/home_shell_router.dart:491]: same as above.

**Review by gnunicorn — APPROVED** [review]
but there is a bug that needs to be fixed. the other things are up for your digression to do..

File: app/lib/features/events/pages/create_event_page.dart (hunk 2 above)
Comment (kumarpalsinh25) [reply app/lib/features/events/pages/create_event_page.dart:413]: Reason is that even after showing space selector, user may close it without selecting any space so in that case spaceId can be null and error message will show over there. I have also shown this use-case in screen recording for reference.

Comment (kumarpalsinh25) [reply app/lib/features/events/pages/create_event_page.dart:413]: Check video at 00:34

File: app/lib/features/events/pages/create_event_page.dart (hunk 4 above)
Comment (kumarpalsinh25) [reply app/lib/features/events/pages/create_event_page.dart:489]: Opss! Thanks for noticing.

File: app/lib/features/events/pages/create_event_page.dart (hunk 2 above)
Comment (gnunicorn) [reply app/lib/features/events/pages/create_event_page.dart:413]: all good. this was a comment of surprise of something new I learnt about the dart language!

Comment (kumarpalsinh25) [reply app/lib/features/events/pages/create_event_page.dart:413]: Ah I see. You always exited about short syntaxes of programming lang.

**Pull request merged**